otter/convergence/service.py

```diff
@@ -286,16 +287,12 @@ def structure_info(path):
                 'group_id': group,
                 'dirty-flag': CONVERGENCE_DIRTY_DIR + '/' + path}
 
-    def got_children(children):
-        dirty_info = map(structure_info, children)
-        num_buckets = len(all_buckets)
-        converging = [
-            info for info in dirty_info
-            if bucket_of_tenant(info['tenant_id'], num_buckets) in my_buckets]
-        return converging
-
-    eff = Effect(GetChildren(CONVERGENCE_DIRTY_DIR))
-    return eff.on(got_children)
+    dirty_info = map(structure_info, divergent_flags)
+    num_buckets = len(all_buckets)
+    converging = [
+        info for info in dirty_info
+        if bucket_of_tenant(info['tenant_id'], num_buckets) in my_buckets]
```

Comment:
Maybe I need to do less lisp, because this is starting to get less legible to me than a filter. (nonactionable)